staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -1626,7 +1626,13 @@ func (az *Cloud) reconcileLoadBalancerRule(
 
 	var expectedProbes []network.Probe
 	var expectedRules []network.LoadBalancingRule
+	highAvailabilityPortsEnabled := false
 	for _, port := range ports {
+		if highAvailabilityPortsEnabled {
+			// Since the port is always 0 when enabling HA, only one rule should be configured.
```

Comment:
> Also it seems like its possible that HA rule might not be the first, in which case you will have more than one rule, if a non HA rule snuck in first.

this shouldn't happen as the HA mode is determined by service annotation, instead of some configure on ports. And, when the first HA rule created, `highAvailabilityPortsEnabled` would be marked as true.

Reply:
> Might be nice to add an info line that you are skipping additional rules.

And this is actually not skipping additional rules. It is one rule to allow all ports, that's why the comment said `only one rule should be configured`.